api/spark.py

```
@@ -111,42 +111,53 @@ def run_spark(user_id: int, chat_id: int, conn, cfg) -> dict:
+                   FROM papers
+                   WHERE NOT processed AND NOT skipped
+                   ORDER BY (
+                     relevance_score + %s * LN(GREATEST(COALESCE(citation_count, 0) + 1, 1))
+                   ) DESC
+                   LIMIT 1""",
+                (cfg.citation_weight,),
+            )
+            paper = cur.fetchone()
+        if paper:
+            return paper
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)